lib/packwerk/cache.rb

```diff
@@ -0,0 +1,131 @@
+# frozen_string_literal: true
+# typed: strict
+
+#
+# There are some known bugs in this cache:
+# 1) The cache should be busted if the contents of `packwerk.yml` change, since custom associations
+# can affect what is considered a violation
```

Comment:
Can we let `packwerk.yml` be part of the cache key, or have a separate cache key for `packwerk.yml` that triggers us to re-parse all source files?

Reply:
Sure, I can add `packwerk.yml` as a cache key. Note that only really covers `custom_associations` (exclude/include paths should not affect the output), and changing of inflections would also invalidate the cache.

If we wanted, I suppose, we could keep the JSONified inflections as a cache input too, and invalidate the cache if that jsonified inflections changes 🤔 

That would make this more "complete," although another part of me is thinking of the tradeoff between additional cache complexity vs. practically how it affects user experience (I imagine inflections won't change too often). I'll defer to you on which approach to take! Happy to do either.

Reply:
Here it is! https://github.com/Shopify/packwerk/pull/169/commits/2335703e7ae80a2e79ebe3e851a877261ae8e513